### ISSUE TITLE
How do I trigger multiple times after connecting to a channel?

### ISSUE THREAD

**Issue by aj-michael** [pusher/pusher-websocket-swift#29]
Apologies in advance, I'm probably missing something obvious.

Currently, I am subscribing to a channel and then immediately triggering several events with the same name. There is a time delay before pusher connects the channel and the result is only the final event is sent.

So to fix this, the best way would be to either queue all of the messages until the channel connects or for me to wait until the channel is connected to begin sending messages. How do I know when the channel is connected?

**Comment by hamchapman** [comment]
What should happen is if you call `yourChannel.trigger(...)` and the channel isn't subscribed to, then the event will get stored in an `unsentEvents` property on the channel.

Then when a successful subscription for said channel does occur, then this bit of code gets run: 

https://github.com/pusher/pusher-websocket-swift/blob/master/Source/PusherSwift.swift#L258-L261

So, in theory, the events should all be being triggered as soon as the connection (and subscription) has succeeded. 

If you've got some example code where that's not happening though, please share it and I'll take a look!

**Comment by aj-michael** [comment]
Yes, I've seen that code. My problem is that you only store one event with the same name (which maybe is intended? I'm new to pusher). Specifically, this line:

https://github.com/pusher/pusher-websocket-swift/blob/master/Source/PusherSwift.swift#L611

The code describing my situation would be something like

```swift
let pusher = Pusher(key: "abc")
let channel = pusher.subscribe("my-channel")
channel.trigger("my-event", ["hello": "world1"])
channel.trigger("my-event", ["hello": "world2"])
```

With this code, `["hello":"world1"]` will never get published.

**Comment by hamchapman** [comment]
Ah I see! Yep, that's a fairly stupid oversight on my part. I hadn't considered that situation.

I'll get a fix for this out by the end of the day.

Thanks for the report!

**Comment by aj-michael** [comment]
Awesome, thanks a lot!

**Comment by hamchapman** [comment]
Think I've got a simple change to fix it. I'm pretty tired right now though so don't fully trust my brain.

I'll take a look with fresh eyes in the morning and then merge and cut a new release: https://github.com/pusher/pusher-websocket-swift/commit/0fedc53bedf1535ce9687c3ac9fc893f0940424d

**Comment by hamchapman** [comment]
Fixed in the just-released 0.2.3

**Comment by aj-michael** [comment]
Awesome, thanks a lot! :+1:
